octopoes/octopoes/api/router.py

```diff
@@ -76,21 +76,24 @@ def settings() -> Settings:
     return Settings()
 
 
-def xtdb_session(client: str = Depends(extract_client), settings_: Settings = Depends(settings)) -> XTDBSession:
-    xtdb_client_ = get_xtdb_client(settings_.xtdb_uri, client, settings_.xtdb_type)
-    return XTDBSession(xtdb_client_)
+def xtdb_session(
+    client: str = Depends(extract_client), settings_: Settings = Depends(settings)
+) -> Generator[XTDBSession, None, None]:
+    session = XTDBSession(get_xtdb_client(settings_.xtdb_uri, client, settings_.xtdb_type))
+
+    try:
+        yield session
+    finally:
+        session.commit()
+        logger.info("Committed XTDBSession")
```

Comment:
The old code also did not commit when an exception happened and I don't think that we want that, so we shouldn't do that inside finally.